woocommerce/woocommerce
# Add variation switcher in footer for variation page

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This adds a new VariationSwitcherFooter component that will be added to the footer of the variation edit screen.
This allows a user to switch between variations pretty seamlessly.

https://github.com/woocommerce/woocommerce/assets/2240960/85bb21cb-1cf7-4b7b-b71b-04a68721c0bd

Closes #40591  .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure feature `New product editor` is enabled under `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`
2. Make sure feature `product-variation-management` is enabled under `Features` tab from `/wp-admin/tools.php?page=woocommerce-admin-test-helper` (WooCommerce Beta Tester plugin) is required
3. Go to `/wp-admin/admin.php?page=wc-admin&path=/add-product&tab=variations` and add some attributes
4. Wait until variations get generated
5. Then publish the product and save its `productId` and one `variationId` you like ( you can find the variation ID by clicking the 3 dot menu of one of the variations )
6. Then go to `/wp-admin/admin.php?page=wc-admin&path=/product/{productId}/variation/{variationId}&tab=general` (replace the `{productId}` and `{variationId}` with the values picked in point 5)
7. A footer should now be present with arrows pointing forward and backward to the previous/next variation.
8. Add an image to the current variation and hit save/publish and click on the next variation using the switcher.
9. Notice how the new variation loads and now the previous variation (with image ) shows up in the switcher footer.
10. While switching also check if the `wcadmin_product_variation_switch_previous` and `wcadmin_product_variation_switch_next` tracks are being fired correctly.
11. Continue to switch through all the variations and try both previous and next, things should work smoothly and the data of the form should update with each variation.
12. Now click the 3 dot menu and click **Delete** which should redirect to the main product.
13. Now go to the `edit` menu of one of the existing variations again and switch through ( it should not be cycling through the deleted version ).
14. Now delete all the variations except for one and go to the edit variation screen
15. The switcher should not be present.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
